aospy/utils/times.py

```diff
@@ -51,6 +55,46 @@ def apply_time_offset(time, years=0, months=0, days=0, hours=0):
                                           days=days, hours=hours))
 
 
+def average_time_bounds(ds):
+    """Return the average of each set of time bounds in the Dataset.
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        A Dataset containing a time bounds array with name matching
+        internal_names.TIME_BOUNDS_STR.  This time bounds array must have two
+        dimensions, one of which's coordinates is the Dataset's time array, and
+        the other is length-2.
+
+    Returns
+    -------
+    xarray.DataArray
+        The mean of the start and end times of each timestep in the original
+        Dataset.
+
+    Raises
+    ------
+    ValueError
+        If the time bounds array doesn't match the shape specified above.
+
+    """
+    bounds = ds[TIME_BOUNDS_STR]
+    if not dict(bounds.sizes) == {TIME_STR: ds.sizes[TIME_STR], BOUNDS_STR: 2}:
+        msg = ("Time bounds array has unsupported shape of {0}.  Time bounds "
+               "must have one dimension corresponding to the Dataset's time "
+               "array and one 'bounds' dimension of length 2.  Length of "
+               "Dataset's time array is {1}.").format(bounds.shape,
+                                                      len(ds[TIME_STR]))
+        raise ValueError(msg)
+
+    start_bounds = bounds.isel(**{BOUNDS_STR: 0})
```

Comment:
The function you've written here is nice, because it can be applied to both dates and floats.  That said, we only use this function on floats (i.e. before we decode times).  Therefore optionally I think it would be possible to simplify the logic (and tests) slightly:
```python
def average_time_bounds(ds):
    bounds = ds[TIME_BOUNDS_STR]
    new_times = bounds.mean(dim=BOUNDS_STR, keep_attrs=True).rename(TIME_STR)
    new_times[TIME_STR] = new_times
    return new_times
```

Reply:
Note the slight edit above (the addition of `.drop`)